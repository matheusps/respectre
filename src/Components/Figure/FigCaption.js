import React from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'

const FigCaption = ({ className, position, children, ...props }) => {
  const figcaptionClasses = classNames(
    className,
    'figure-caption',
    `text-${position}`
  )
  return <figcaption className={figcaptionClasses}>{children}</figcaption>
}

FigCaption.propTypes = {
  /** Aditional css classes */
  className: PropTypes.string,
  /** Caption position */
  position: PropTypes.oneOf(['center', 'left', 'right', 'justify']),
}

FigCaption.defaultProps = {
  position: 'center',
}

export default FigCaption
