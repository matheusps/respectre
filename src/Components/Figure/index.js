import React from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'

import Image from '../Image'
import FigCaption from './FigCaption'

const Figure = ({ className, children }) => {
  const figureClasses = classNames(className, 'figure')
  return <figure className={figureClasses}>{children}</figure>
}

Figure.Image = Image
Figure.Caption = FigCaption

Figure.propTypes = {
  /** Aditional css classes */
  className: PropTypes.string,
}

export default Figure
