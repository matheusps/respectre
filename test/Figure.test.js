import React from 'react'
import renderer from 'react-test-renderer'
import Figure from '../src/Components/Figure'

describe('Figure', () => {
  it('should match snapshot', () => {
    const component = renderer.create(<Figure />).toJSON()
    expect(component).toMatchSnapshot()
  })

  it('should match snapshot with image', () => {
    const component = renderer
      .create(
        <Figure>
          <Figure.Image src="testing" />
        </Figure>
      )
      .toJSON()
    expect(component).toMatchSnapshot()
  })

  it('should match snapshot with caption', () => {
    const component = renderer
      .create(
        <Figure>
          <Figure.Caption>Testing</Figure.Caption>
        </Figure>
      )
      .toJSON()
    expect(component).toMatchSnapshot()
  })

  it('should match snapshot with image and caption', () => {
    const component = renderer
      .create(
        <Figure>
          <Figure.Image src="testing" />
          <Figure.Caption>Testing</Figure.Caption>
        </Figure>
      )
      .toJSON()
    expect(component).toMatchSnapshot()
  })
})
