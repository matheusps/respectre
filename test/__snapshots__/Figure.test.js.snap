// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Figure should match snapshot 1`] = `
<figure
  className="figure"
/>
`;

exports[`Figure should match snapshot with caption 1`] = `
<figure
  className="figure"
>
  <figcaption
    className="figure-caption text-center"
  >
    Testing
  </figcaption>
</figure>
`;

exports[`Figure should match snapshot with image 1`] = `
<figure
  className="figure"
>
  <img
    className="img-fit-undefined"
    src="testing"
  />
</figure>
`;

exports[`Figure should match snapshot with image and caption 1`] = `
<figure
  className="figure"
>
  <img
    className="img-fit-undefined"
    src="testing"
  />
  <figcaption
    className="figure-caption text-center"
  >
    Testing
  </figcaption>
</figure>
`;
